Lib/zipfile/__init__.py

```diff
@@ -1010,7 +1012,9 @@ def _update_crc(self, newdata):
         self._running_crc = crc32(newdata, self._running_crc)
         # Check the CRC if we're at the end of the file
         if self._eof and self._running_crc != self._expected_crc:
-            raise BadZipFile("Bad CRC-32 for file %r" % self.name)
+            raise BadZipFile(
+                    f"Bad CRC-32 for file {self.name!r} expected "
+                    f"{hex(self._expected_crc)} != {hex(self._running_crc)}")
```

Comment:
Did you keep the beginning of the error message identical to keep compatibility?
If not, then it could be easier to read if it was like `bad crc32 {got value} for file {name}, expected {expected value}`. For now it’s slightly unpleasant to read about bad value, expected, expected, bad.